openedx/core/djangoapps/plugins/constants.py

```diff
@@ -76,3 +76,19 @@ class PluginSignals(object):
 
     RELATIVE_PATH = u'relative_path'
     DEFAULT_RELATIVE_PATH = u'signals'
+
+
+class PluginContextsViews(object):
+    STUDENT_DASHBOARD = u'student_dashboard'
```

Comment:
**Warning (and request):** Let's not have the plugin-framework level code have any feature-dependent code in it.  In fact, I would like to extract [openedx/core/djangoapps/plugins](https://github.com/edx/edx-platform/tree/master/openedx/core/djangoapps/plugins) to its own repo so other Django Projects can use this framework.

**Proposal:** Have this constant be defined in a doc/README file within the `student_dashboard` app so plugins of the student dashboard know what the string value is.  Remember, the plugins would be defined outside of the dashboard code and would unlikely python-import the dashboard's constants anyway.  What do you think?

Reply:
Hmm. My hopes was that this would serve as both the list of all views that are context-pluggable while also allowing us to change names in one place if the need arises since all external apps could pull from this single source. I'm not entirely against documented named strings, it lies in that field of more flexible / more breakable.